cxplat/src/cxplat_winuser/memory_winuser.cpp

```diff
@@ -165,31 +170,36 @@ __drv_allocatesMem(Mem) _Must_inspect_result_ _Ret_writes_maybenull_(new_size) v
 }
 
 void
-cxplat_free(_Frees_ptr_opt_ void* pointer)
+cxplat_free_any_tag(_Frees_ptr_opt_ void* pointer)
 {
-    if (pointer == nullptr)
-    {
+    if (pointer == nullptr) {
         return;
     }
     if (_cxplat_leak_detector_ptr) {
         _cxplat_leak_detector_ptr->unregister_allocation(reinterpret_cast<uintptr_t>(pointer));
     }
     cxplat_allocation_header_t* header = _header_from_pointer(pointer);
-    if (header->pool_type == CxPlatNonPagedPoolNxCacheAligned)
-    {
+    if (header->pool_type == CxPlatNonPagedPoolNxCacheAligned) {
         uint8_t* memory_block = _memory_block_from_aligned_pointer(pointer);
         _aligned_free(memory_block);
-    }
-    else
-    {
+    } else {
         uint8_t* memory_block = _memory_block_from_unaligned_pointer(pointer);
         free(memory_block);
     }
 }
 
+void
+cxplat_free(_Frees_ptr_opt_ void* pointer, uint32_t tag)
+{
+    if (pointer != nullptr) {
+        cxplat_allocation_header_t* header = _header_from_pointer(pointer);
```

Comment:
For a separate PR, but this allocation header should be debug only.

Reply:
Agree, filed #111